optax/_src/transform.py

```diff
@@ -351,7 +353,16 @@ def update_fn(updates, state, params=None):
     mu = update_moment(updates, state.mu, b1, 1)
     nu = update_moment_per_elem_norm(updates, state.nu, b2, 2)
     count_inc = numerics.safe_int32_increment(state.count)
-    mu_hat = bias_correction(mu, b1, count_inc)
+    if nesterov:
+      mu_hat = jax.tree_util.tree_map(
+          lambda m, g: b1 * m + (1 - b1) * g,
+          bias_correction(mu, b1, numerics.safe_int32_increment(count_inc)),
+          bias_correction(updates, b1, count_inc))
+      # Dozat 2016 https://openreview.net/pdf?id=OM0jvwB8jIp57ZJjtNEZ
+      # Algorithm 2 further multiplies Adam's standard nu_hat by b2. It is
+      # unclear why. Other Nadam implementations also omit the extra 2 factor.
+    else:
+      mu_hat = bias_correction(mu, b1, count_inc)
     nu_hat = bias_correction(nu, b2, count_inc)
```

Comment:
```suggestion
    else:
      mu_hat = bias_correction(mu, b1, count_inc)
    # Dozat 2016 https://openreview.net/pdf?id=OM0jvwB8jIp57ZJjtNEZ
    # Algorithm 2 further multiplies Adam's standard nu_hat by b2. It is
    # unclear why. Other Nadam implementations also omit the extra 2 factor.
    nu_hat = bias_correction(nu, b2, count_inc)
```

Reply:
accepted suggestion :-)

will be visible in a couple minutes